File: Pumbank/src/main/java/com/pumbank/models/Paga.java
package com.pumbank.models;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name="paga")

public class Paga {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private int pgid;

	@Column(name="cantidad")
	private double cantidad;
	
	@Column(name="frecuencia")
	private int frecuencia;
	
	@Column(name="padre")
	private int pid;
	
	@Column(name="hijo")
	private int hid;
	
	public Paga() {}
	
	
	public Paga(int pgid, double cantidad, int frecuencia, int pid, int hid) {
		super();
		this.pgid = pgid;
		this.cantidad = cantidad;
		this.frecuencia = frecuencia;
		this.pid = pid;
		this.hid = hid;
	}

	public int getPgid() {
		return pgid;
	}

	public void setPgid(int pgid) {
		this.pgid = pgid;
	}


	public double getCantidad() {
		return cantidad;
	}

	public void setCantidad(double cantidad) {
		this.cantidad = cantidad;
	}


	public int getFrecuencia() {
		return frecuencia;
	}

	public void setFrecuencia(int frecuencia) {
		this.frecuencia = frecuencia;
	}

	public int getPid() {
		return pid;
	}

	public void setPid(int pid) {
		this.pid = pid;
	}

	public int getHid() {
		return hid;
	}

	public void setHid(int hid) {
		this.hid = hid;
	}

	public boolean validate() {
		boolean isValid=true;
		
		if(this.cantidad<0) isValid=false;
		if(this.frecuencia<=0) isValid=false;
		
		return isValid;
	}

	
}

File: my-pumbank/src/app/modelos/Paga.ts
export class Paga { 

    constructor(private _pgid: number,
        private _cantidad: number,
        private _frecuenciadias: number,
        private _pid: number,
        private _hid: number,
        ){
    }

    get pgid() {return this._pgid;}
    get cantidad() {return this._cantidad;}
    get frecuenciadias() {return this._frecuenciadias;}
    get pid() {return this._pid;}
    get hid() {return this._hid;}

    set pgid(pgid) {this._pgid = pgid;}
    set cantidad(cantidad) {this._cantidad = cantidad;}
    set frecuenciadias(frecuenciadias) {this._frecuenciadias = frecuenciadias;}
    set pid(pid) {this._pid = pid;}
    set hid(hid) {this._hid = hid;}

}
